packages/collections/README.md

```diff
@@ -1,11 +1,383 @@
 # `collections`
 
-> TODO: description
+The `collections` package provides functions to work with lists in various ways.
 
-## Usage
+## Functions
 
+### `bifurcateWhen`
+
+This function splits an array once the provided function evaluates to `true` for an element of that array. The element that resulted in the `true` evaluation is the first element of the second returned array.
+
+```ts
+import { bifurcateWhen } from '@execonline-inc/collections';
+
+const list = [1, 2, 3, 4, 5];
+const result = bifurcateWhen((item: number): boolean => item === 3, list);
+// [[1, 2], [3, 4, 5]]
+```
+
+### `concat`
+
+This curried function simply concatenates two arrays.
+
+```ts
+import { concat } from '@execonline-inc/collections';
+
+const listA = [1, 2, 3];
+const listB = [4, 5, 6];
+
+const result = concat(listA)(listB);
+// [1, 2, 3, 4, 5, 6]
+```
+
+### `concatAll`
+
+This function takes an array of arrays, and concatenates them all in order.
+
+```ts
+import { concatAll } from '@execonline-inc/collections';
+
+const lists = [
+  [1, 2, 3],
+  [4, 5, 6],
+  [7, 8, 9],
+];
+
+const result: number[] = concatAll(lists);
+// [1, 2, 3, 4, 5, 6, 7, 8, 9]
+```
+
+### `dropUntil`
+
+This function has a curried and non-curried form. It takes an evaluation function and an array.
+
+The evaluation function is called on elements of the array until it evaluates to `true`. The element that caused the function to evaluate to `true` and subsequent elements of the array are returned in a new array.
+
+```ts
+import { dropUntil } from '@execonline-inc/collections';
+
+const list = [1, 2, 3, 4, 5];
+const result: number[] = dropUntil((item: number): boolean => item === 3)(list);
+// [3, 4, 5]
+```
+
+### `uniqBy`
+
+This curried function takes a transformation function and an array. Each element of the array is passed to the transformation function. Duplicate evaluations of this function exlude the corresponding original element from the returned array.
+
+```ts
+import { uniqBy } from '@execonline-inc/collections';
+
+const list = [1, 2, 3, 4, 5];
+const result: number[] = uniqBy((item: number): string => (item > 3 ? String(item) : ''));
+// [1, 4, 5]
+```
+
+### `uniq`
+
+This function returns only the unique elements of a given array.
+
+```ts
+import { uniq } from '@execonline-inc/collections';
+
+const list = [1, 1, 2, 3, 3];
+const result: number[] = uniq(list);
+// [1, 2, 3]
+```
+
+### `find`
+
+This function has a curried and non-curried form. It takes an evaluation function and an array.
```

Comment:
`This function has a curried and non-curried form.` should probably be that last paragraph about this function. 

I think the documentation should be written more like this:
- Summary of feature
- Details of how it works
- Curried and uncurried, or other minutiae as appropriate.

```
`find` returns the first matching element in an Array.

Matching is determined by the function passed as the first argument. 
If it returns true, then a match has been identified.

If a match is found, the item is return wrapped in a `Just`. Otherwise, a `Nothing` is returned.

This function has a curried and non-curried form.
```

Reply:
Did we ever make this documentation change?

Reply:
No, haven't had a chance.